src/locale/helpers.ts

```diff
@@ -31,9 +32,37 @@ export function code3ToCode2Strict(lang: string): string | undefined {
   return undefined
 }
 
-export function codeToLanguageName(lang: string): string {
-  const lang2 = code3ToCode2(lang)
-  return LANGUAGES_MAP_CODE2[lang2]?.name || lang
+function getLocalizedLanguage(
+  langCode: string,
+  appLang: string,
+): string | undefined {
+  const allNames = new Intl.DisplayNames([appLang], {
+    type: 'language',
+    fallback: 'none',
+    languageDisplay: 'standard',
+  })
+  const translatedName = allNames.of(langCode)
```

Comment:
[of()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/of) expects a language code matching the [unicode_language_id](https://unicode.org/reports/tr35/#Unicode_language_identifier) rule from CLDR which is only a subset of BCP 47, the format permitted in AT Protocol. E.g. `zh-YUE` is a valid BCP 47 code but will make the `of()` method throw a `RangeError`, so the call needs to be wrapped in a `try`/`catch` block.

Reply:
Fixed, good catch! Thanks :)